src/mpl/src/gpu/mpl_gpu_fallback.c

```diff
@@ -0,0 +1,37 @@
+/*
+ *  Copyright (C) by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ */
+
+#include "mpl.h"
+
+MPL_SUPPRESS_OSX_HAS_NO_SYMBOLS_WARNING;
+
+#ifdef MPL_HAVE_GPU_FALLBACK
```

Comment:
This is what I'm saying we should not have.

Reply:
The fallback implements the MPL wrapper functions declared in `mpl_gpu.h`. But the fallback implementation should not be built when there is a GPU detected. Otherwise, we would have multiple definitions of the same function.

Reply:
Here's what I was recommending:

```
#ifdef MPL_HAVE_CUDA
#include "mpl_gpu_cuda.h"
#elif defined (MPL_HAVE_XE)
#include "mpl_gpu_xe.h"
#elif defined (MPL_HAVE_HIP)
#include "mpl_gpu_hip.h"
#else
#include "mpl_gpu_fallback.h"
#endif
```

Same thing in the `Makefile.mk` if `.c` files are needed (although it doesn't look like they are in this case).

Reply:
This should be fixed now. Added a similar branch in `src/gpu/Makefile.mk` to handle the conditional build.